Fixed demos for Progress Indicator and Modal Dialog in web

tritland opened the pull request:
Worked on both Progress Indicator and Modal Dialog demos as they share show/hide modal functionality and styling.

Commit:
fixed styling for progress bar and launch spinner button

Commit:
fixed styling for progress bar and launch spinner button

Commit:
fixed show and hide modal functionality for demos on progress indicator and modal dialog pages for web

Commit:
fixed show and hide modal functionality for demos on progress indicator and modal dialog pages for web

Commit:
fixed inline dynamic loading waiting example demo for web progress indicator

Commit:
fixed inline dynamic loading waiting example demo for web progress indicator

marshcooke reviewed (changes requested):
These look super great, Tom! Only a few things need to be updated: 
- Web > Progress Indicator > in dark mode, the modal needs to have dark mode applied to the background (will we need a different color combo for the loading spinner that populates?) 
- Web > Modal Dialog > in dark mode, the modal needs to have dark mode applied to the background (the text is already displaying in white)

Commit:
added support for dark mode and removed some comments

marshcooke reviewed (dismissed):
This looks good to me, great work, Tom!

silverli committed:
fix: refactoring some modal progressscss styles

silverli committed:
fix: refactoring some modal progressscss styles

marshcooke reviewed (changes requested):
All looks great! The only thing I noticed when re-reviewing today, is that in Safari, the focus indicator is not present on the Modal Dialog > opened example modal > close button.

silverli commented on public/content/documentation/web/component/modal-dialog.md:
````diff
@@ -117,21 +117,25 @@ Since browser support for `<dialog>` is incomplete, if you plan to use the `<dia
 
 ```html
-   <button id="showModal">
+   <button data-fn="showModal" 
````
ensure your example code and the live code match relatively closely (example doesn't have to contain all the clasess, etc.). I was noticing some discrepancies

silverli commented on src/utils/markdownFunctions.ts:
```diff
@@ -200,4 +200,186 @@ export const getMarkdownFunctionMap = (
+  }
+
+  // Ensure ARIA
+  if (!dlg.getAttribute('role')) dlg.setAttribute('role', 'dialog');
```
these attributes should just be set up in the HTML originally, not by the JS

silverli commented on src/utils/markdownFunctions.ts:
```diff
@@ -200,4 +200,186 @@ export const getMarkdownFunctionMap = (
     window.scrollTo({ top: 0, behavior: 'smooth' });
   },
+
+// show/close modal functionality
```
This code is incredibly hard to read. 

The variables are not clearly named. 

I worry that there is A LOT more code than is needed here, but honestly I cannot tell what is being done in most of this. 

All that is need for a modal being hidden/closed should be a toggling of a class and moving .focus.

And then for when the modal is open, we just need focus to be contained inside the modal which I'm not seeing where the code/logic is here. 

I recommend going off Charlie's code from the old project: https://github.com/tmobile/magentaA11y/blob/main-copy/assets/scripts/index.js#L97 

OR using JS  adapted from the WAI-ARIA example: https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/

silverli replied:
Charlie's trapfocus code is here: https://github.com/tmobile/magentaA11y/blob/main-copy/assets/scripts/index.js#L598

silverli reviewed (changes requested):
Hey @tritland  I finally was able to put my thoughts down as to what I want to see refactored here. 

This is a good start, but I want to see closer conformance to project coding standards, use of CSS variables, and I also want to see cleaner/more performant code. 

I'm going to rebase this for you so you don't have to do that, so when you pull it down again, you can do: `git pull origin fix-progress-indicator-js --rebase`

tritland committed:
Revisions to modal dialog example

tritland committed:
Revisions to modal dialog example

tritland committed:
updates for progress indicator example scripts and styles

tritland committed:
updates for progress indicator example scripts and styles

marshcooke reviewed (dismissed):
This all looks good to me, but I know @silverli will want to go through this more thoroughly and resolve some of her previous comments. 

I can confirm it works in light and dark modes, as well as passes keyboard and screen reader testing - all looks great!

MyLightIsOn committed:
Merge remote-tracking branch 'origin/main' into fix-progress-indicator-js

MyLightIsOn committed:
refactor: Restored code that accidentally got removed during the merge conflict resolution.

MyLightIsOn committed:
refactor: Restored code that accidentally got removed during the merge conflict resolution.

MyLightIsOn committed:
refactor: Simplified SCSS for progress indicators and aligned color variables with design tokens

MyLightIsOn committed:
refactor: Simplified SCSS for progress indicators and aligned color variables with design tokens

MyLightIsOn committed:
refactor: Removed redundant aria-label="Continue" from the Continue button and updated SCSS backdrop color format

MyLightIsOn committed:
refactor: Removed redundant aria-label="Continue" from the Continue button and updated SCSS backdrop color format

MyLightIsOn committed:
feat: Add progress chip SCSS styles and modal/dialog helper functions

- Introduced inline progress chip styles with animations in `_example-progress-indicator.scss`.
- Added functions to handle modals: `showModal`, `closeModal`, and `startProgress` with auto-close and progress simulation logic.

MyLightIsOn committed:
feat: Add progress chip SCSS styles and modal/dialog helper functions

- Introduced inline progress chip styles with animations in `_example-progress-indicator.scss`.
- Added functions to handle modals: `showModal`, `closeModal`, and `startProgress` with auto-close and progress simulation logic.

MyLightIsOn committed:
fix: Add focus-visible outline to modal close button for Safari

Safari does not show the default browser focus ring on buttons, so an
explicit :focus-visible style is needed for keyboard accessibility.

MyLightIsOn committed:
fix: Add focus-visible outline to modal close button for Safari

Safari does not show the default browser focus ring on buttons, so an
explicit :focus-visible style is needed for keyboard accessibility.

silverli commented on src/styles/md-styles/_example-progress-indicator.scss:
```diff
@@ -0,0 +1,152 @@
+      content: '';
+      display: block;
+      background-color: var(--color-background);
+      width: calc(100px);
```
calc not needed

silverli commented on src/utils/markdownFunctions.ts:
```diff
@@ -43,6 +43,77 @@ export const getMarkdownFunctionMap = (
+   * Opens a modal dialog identified by the button's data-target attribute.
+   * Supports auto-close via data-auto-close (milliseconds).
+   */
+  showModal: (event) => {
```
we should call this fn something else as there is a JS native function referenced below called .showModal()

silverli commented on src/utils/markdownFunctions.ts:
```diff
@@ -43,6 +43,77 @@ export const getMarkdownFunctionMap = (
+    const dialog = document.getElementById(targetId) as HTMLDialogElement;
+    if (!dialog) return;
+    dialog.showModal();
+    const autoClose = button.getAttribute('data-auto-close');
```
not sure we should auto close like this... goes against WCAG timing

silverli commented on src/utils/markdownFunctions.ts:
```diff
@@ -43,6 +43,77 @@ export const getMarkdownFunctionMap = (
+    if (container) container.setAttribute('aria-busy', 'true');
+    chip.removeAttribute('hidden');
+
+    const steps: Array<[string, number]> = [
```
let's split this part into another function if possible (or call it conditionally if a chip exists)

silverli replied:
I say that because ariabusy should be toggled for the other progressbar example as well

silverli replied:
let's do this later

silverli commented on src/styles/md-styles/_example-modal-dialog.scss:
```diff
@@ -0,0 +1,52 @@
+      width: 2rem;
+      height: 2rem;
+
+      &:focus-visible {
```
delete this. we dont use custom focus

silverli commented on public/content/documentation/web/component/modal-dialog.md:
```diff
@@ -117,21 +117,25 @@ Since browser support for `<dialog>` is incomplete, if you plan to use the `<dia
-         id="modal"
+         id="dialog-example-modal"
+         class="dialog"
          tabindex="-1"
```
one concern to deal with later: 

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/dialog#additional_notes):

> Do not add the tabindex property to the <dialog> element as it is not interactive and does not receive focus. The dialog's contents, including the close button contained in the dialog, can receive focus and be interactive.

silverli replied:
I wrote this down to track elsewhere

silverli committed:
merge conflict and linting

silverli committed:
fix: a couple minor edits - remove custom focus, remove autoClose fn, and some scss linting
